docs/tutorials/federation/set-up-coredns-provider-federation.md

```diff
@@ -0,0 +1,126 @@
+---
+assignees:
+- shashidharatd
+title: Setting up CoreDNS as DNS provider for Cluster Federation
+---
+
+* TOC
+{:toc}
+
+This guide explains how to configure and deploy CoreDNS to be used as
+DNS provider for Cluster Federation.
+
+CoreDNS can be deployed in various configuration suiting the cluster
+federation deployment. In this guide we are explaining CoreDNS
+deployment as a reference and can be tweaked to suit the needs of the
+platform and the cluster federation.
+
+## Prerequisites
+
+This guide assumes that you have a running Kubernetes cluster (which is
+referenced as host cluster). Please see one of the
+[getting started](/docs/getting-started-guides/) guides for
+installation instructions for your platform.
+
+## Deploying CoreDNS and etcd charts
+
+To deploy CoreDNS we shall make use of helm charts. CoreDNS will be
+deployed with [etcd](https://coreos.com/etcd) as the backend and should
+be pre-installed. etcd also can be deployed using helm charts. Shown
+below are the instructions to deploy etcd.
+
+```shell
+$ helm install --namespace my-namespace --name etcd-operator stable/etcd-operator
+$ helm upgrade --namespace my-namespace --set cluster.enabled=true etcd-operator stable/etcd-operator
+```
+
+> Note: etcd default deployment configurations can be overridden, suiting the
+host cluster.
+
+After deployment succeeds, etcd can be accessed with
+[http://etcd-cluster.my-namespace:2379](http://etcd-cluster.my-namespace:2379) endpoint within the host cluster.
+
+CoreDNS default configuration should be customized to suit federation
```

Comment:
add "the": "`The` CoreDNS default..." and "...customized to suit `the` federation."